### ISSUE TITLE
AssaultCube is "too dark" in general

### ISSUE THREAD

**Issue by drian0** [assaultcube/AC#312]
On the vast majority of devices AssaultCube is too dark. You can not see your opponents well. 

The current workarounds to mitigate this problem are the "gamma" and "lightscale" settings. However both of them are problematic because they result in poor contrasts or poor visual representation. 

A sustainable solution would be to go through all official maps and re-light them properly. For more information please see also #285

**Comment by Felix-The-Ghost** [comment]
You might be right with your theory that at some point most of the mappers were already using lightscale when they created their maps, perpetuating dark maps (on default settings)

I'm actually curious now if I'm guilty of that; I don't remember that command so maybe the map I made **ac_district** is too bright for people using /lightscale or too dark for default settings.

**Comment by BrutalSystem** [comment]
> You might be right with your theory that at some point most of the mappers were already using lightscale when they created their maps, perpetuating dark maps (on default settings)
> 
> I'm actually curious now if I'm guilty of that; I don't remember that command so maybe the map I made **ac_district** is too bright for people using /lightscale or too dark for default settings.

apparently we found a solution /ambient

**Comment by Felix-The-Ghost** [comment]
> apparently we found a solution; check /ambient

I can't. Guessing this is a command that sets the minimum brightness though?

Funny enough I once implemented exactly that, though it was mainly for map editing as I didn't like /fullbright

**Comment by Felix-The-Ghost** [comment]
Yep, exactly what I thought. I wonder if any maps would be negatively affected (areas that are intended to be very dark)

**Comment by cf-1984** [comment]
> > I can't. Guessing this is a command that sets the minimum brightness though?
> 
> Controls the ambient lighting of the map, i.e. how bright areas not affected by any light entities will appear.

Ambient does affect all verts, regardless if those are already affected by light entities or not. Therefore, yes Felix is right, you set the minimum brightness.

> Yep, exactly what I thought. I wonder if any maps would be negatively affected (areas that are intended to be very dark)

Well, the idea is getting rid off of those very dark areas. However, if someone intentionally left areas darker, where players won't be able to move anyway, than this might indeed not be the best approach. Atm I couldn't think of any map in the official stock where this might become an issue, though.

However, the return value when calling /ambient omitting the parameter is messed up big time. Seems like it always only returns the last value that has been set during a session. Is there an open issue for that already?

**Comment by BrutalSystem** [comment]
> However, the return value when calling /ambient omitting the parameter is messed up big time. Seems like it always only returns the last value that has been set during a session. Is there an open issue for that already?

https://github.com/assaultcube/AC/issues/345 I don't know exactly if that is what you mean.

**Comment by cf-1984** [comment]
> #345 I don't know exactly if that is what you mean.

Yes, exactly that. :)
